laikaboss/objectmodel.py

```diff
@@ -50,6 +50,25 @@ def ensureNotUnicode(buffer):
     else:
         return buffer
 
+# Utility function to make sure the buffer is a string and not None 
+# (or whatever other weirdness comes through)
+def ensureStr(child_buffer):
+    # buffers and bytearrays can be cast to string
+    try:
+        if type(child_buffer) == buffer or type(child_buffer) == bytearray:
+            child_buffer = str(child_buffer)
+    except:
+        # Test cases do not produce any exceptions, but it's here just in case
+        raise Exception("Buffer of %s found, not creating child scanObject" % str(type(child_buffer)))
+
+    # refuse to process anything else, as non-string objects can crash the worker
+    if not type(child_buffer) is str:
+        raise Exception("Buffer of %s found, not creating child scanObject" % str(type(child_buffer)))
+    
+    child_buffer = ensureNotUnicode(child_buffer)
```

Comment:
I think you have a logic error here. This function will never get called for a unicode object because you're raising an exception above. Therefore instead of a unicode object getting encoded to a utf-8 string, you'd get an exception instead. Can you address this?